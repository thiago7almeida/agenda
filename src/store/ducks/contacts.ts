import {Reducer} from 'redux';

export const Types = {
  // LOGIN: 'login/LOGIN',
};
interface Address {
  cep: string;
  state: string;
  city: string;
  street: string;
  neighborhood: string;
}

export interface Contact {
  name: string;
  phone: string;
  address: Address;
}
export interface ContactsState {
  data: Contact[];
}

const initialState: ContactsState = {
  data: [
    {
      name: 'name',
      phone: '(16) 9963894106',
      address: {
        cep: '05010000',
        state: 'SP',
        city: 'São Paulo',
        street: 'Rua Caiubí',
        neighborhood: 'Perdizes',
      },
    },
    {
      name: 'thiago',
      phone: '(16) 9963894106',
      address: {
        cep: '05010000',
        state: 'SP',
        city: 'São Paulo',
        street: 'Rua Caiubí',
        neighborhood: 'Perdizes',
      },
    },
    {
      name: 'Jose',
      phone: '(16) 9963894106',
      address: {
        cep: '05010000',
        state: 'SP',
        city: 'São Paulo',
        street: 'Rua Caiubí',
        neighborhood: 'Perdizes',
      },
    },
    {
      name: 'Valmir',
      phone: '(16) 9963894106',
      address: {
        cep: '05010000',
        state: 'SP',
        city: 'São Paulo',
        street: 'Rua Caiubí',
        neighborhood: 'Perdizes',
      },
    },
    {
      name: 'Gustavo',
      phone: '(16) 9963894106',
      address: {
        cep: '05010000',
        state: 'SP',
        city: 'São Paulo',
        street: 'Rua Caiubí',
        neighborhood: 'Perdizes',
      },
    },
    {
      name: 'Gabi',
      phone: '(16) 9963894106',
      address: {
        cep: '05010000',
        state: 'SP',
        city: 'São Paulo',
        street: 'Rua Caiubí',
        neighborhood: 'Perdizes',
      },
    },
    {
      name: 'Cristina',
      phone: '(16) 9963894106',
      address: {
        cep: '05010000',
        state: 'SP',
        city: 'São Paulo',
        street: 'Rua Caiubí',
        neighborhood: 'Perdizes',
      },
    },
    {
      name: 'Patricia',
      phone: '(16) 9963894106',
      address: {
        cep: '05010000',
        state: 'SP',
        city: 'São Paulo',
        street: 'Rua Caiubí',
        neighborhood: 'Perdizes',
      },
    },
    {
      name: 'Priscila',
      phone: '(16) 9963894106',
      address: {
        cep: '05010000',
        state: 'SP',
        city: 'São Paulo',
        street: 'Rua Caiubí',
        neighborhood: 'Perdizes',
      },
    },
    {
      name: 'thiago',
      phone: '(16) 9963894106',
      address: {
        cep: '05010000',
        state: 'SP',
        city: 'São Paulo',
        street: 'Rua Caiubí',
        neighborhood: 'Perdizes',
      },
    },
    {
      name: 'Jose',
      phone: '(16) 9963894106',
      address: {
        cep: '05010000',
        state: 'SP',
        city: 'São Paulo',
        street: 'Rua Caiubí',
        neighborhood: 'Perdizes',
      },
    },
    {
      name: 'Valmir',
      phone: '(16) 9963894106',
      address: {
        cep: '05010000',
        state: 'SP',
        city: 'São Paulo',
        street: 'Rua Caiubí',
        neighborhood: 'Perdizes',
      },
    },
    {
      name: 'Gustavo',
      phone: '(16) 9963894106',
      address: {
        cep: '05010000',
        state: 'SP',
        city: 'São Paulo',
        street: 'Rua Caiubí',
        neighborhood: 'Perdizes',
      },
    },
    {
      name: 'Gabi',
      phone: '(16) 9963894106',
      address: {
        cep: '05010000',
        state: 'SP',
        city: 'São Paulo',
        street: 'Rua Caiubí',
        neighborhood: 'Perdizes',
      },
    },
    {
      name: 'Cristina',
      phone: '(16) 9963894106',
      address: {
        cep: '05010000',
        state: 'SP',
        city: 'São Paulo',
        street: 'Rua Caiubí',
        neighborhood: 'Perdizes',
      },
    },
    {
      name: 'Patricia',
      phone: '(16) 9963894106',
      address: {
        cep: '05010000',
        state: 'SP',
        city: 'São Paulo',
        street: 'Rua Caiubí',
        neighborhood: 'Perdizes',
      },
    },
    {
      name: 'Priscila',
      phone: '(16) 9963894106',
      address: {
        cep: '05010000',
        state: 'SP',
        city: 'São Paulo',
        street: 'Rua Caiubí',
        neighborhood: 'Perdizes',
      },
    },
    {
      name: 'Deh',
      phone: '(16) 9963894106',
      address: {
        cep: '05010000',
        state: 'SP',
        city: 'São Paulo',
        street: 'Rua Caiubí',
        neighborhood: 'Perdizes',
      },
    },
  ],
};

const reducer: Reducer<ContactsState> = (state = initialState, action) => {
  switch (action.type) {
    // case Types.LOGIN:
    //   return {...state, isLoged: true};

    default:
      return state;
  }
};
export default reducer;
