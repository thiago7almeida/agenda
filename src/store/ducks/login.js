export const Types = {
  LOGIN: "login/LOGIN",
};

const initialState = {
  isLoged: false,
};

export default function reducer(state = initialState, action) {
  switch (action.type) {
    case Types.LOGIN:
      return { ...state, isLoged: true };

    default:
      return state;
  }
}

export function login() {
  return {
    type: Types.LOGIN,
  };
}
