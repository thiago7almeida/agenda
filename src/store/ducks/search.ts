import {Reducer} from 'redux';

export const Types = {
  SET_NAME: 'search/SET_NAME',
  CLEAR_NAME: 'search/CLEAR_NAME',
};

export interface SearchState {
  name: string;
}

//mock data
const initialState: SearchState = {
  name: '',
};

const reducer: Reducer<SearchState> = (state = initialState, action) => {
  switch (action.type) {
    case Types.SET_NAME:
      return {...state, name: action.payload};
    case Types.CLEAR_NAME:
      return initialState;

    default:
      return state;
  }
};
export default reducer;

export const setName = (name: string) => ({
  type: Types.SET_NAME,
  payload: name,
});
export const clearName = () => ({
  type: Types.SET_NAME,
});
