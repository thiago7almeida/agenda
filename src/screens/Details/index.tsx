import React from 'react';
import {TopNavigation, Divider, Button, Icon} from '@ui-kitten/components';
import {SafeAreaView} from 'react-native-safe-area-context';
import {useNavigation} from '@react-navigation/native';
import {StackScreenProps} from '@react-navigation/stack';

import styles from './styles';
import Form from './components/Form';
import {Contact} from '../../store/ducks/contacts';

type RootStackParamList = {
  Details: {isNew: boolean; item?: Contact};
};

type Props = StackScreenProps<RootStackParamList, 'Details'>;
const IconPlus = () => {
  return (
    <Icon name="arrow-ios-back-outline" fill="black" style={styles.iconPlus} />
  );
};

const ButtonLeft = () => {
  const navigation = useNavigation();
  return (
    <Button
      appearance="ghost"
      onPress={() => navigation.goBack()}
      accessoryLeft={IconPlus}
    />
  );
};

const App = ({route}: Props) => {
  const {isNew, item} = route.params;
  return (
    <SafeAreaView style={styles.container}>
      <TopNavigation
        title="Detalhe do Contato"
        alignment={'center'}
        accessoryLeft={ButtonLeft}
      />
      <Divider />
      <Form isNew={isNew} item={item} />
    </SafeAreaView>
  );
};

export default App;
