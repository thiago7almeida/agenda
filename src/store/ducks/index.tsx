import {combineReducers} from 'redux';
import login from './login';

const reducers = combineReducers({
  login,
});

const rootReducer = (state: any, action: any) => {
  return reducers(state, action);
};
export default rootReducer;
