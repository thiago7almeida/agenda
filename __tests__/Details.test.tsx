import React from 'react';

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Details from '../src/screens/Details';
import {Contact} from '../src/store/ducks/contacts';
import {StackScreenProps} from '@react-navigation/stack';

type RootStackParamList = {
  Details: {isNew: boolean; item?: Contact};
};

type Props = StackScreenProps<RootStackParamList, 'Details'>;

describe('Details', () => {
  describe('rendering', () => {
    it('should match to snapshot - Primary', () => {
      const props: Props = {
        route: {params: {isNew: true}, key: 'Details', name: 'Details'},
        navigation: {
          addListener: jest.fn(),
          canGoBack: jest.fn(),
          dangerouslyGetParent: jest.fn(),
          dangerouslyGetState: jest.fn(),
          dispatch: jest.fn(),
          goBack: jest.fn(),
          isFocused: jest.fn(),
          navigate: jest.fn(),
          pop: jest.fn(),
          popToTop: jest.fn(),
          push: jest.fn(),
          removeListener: jest.fn(),
          replace: jest.fn(),
          reset: jest.fn(),
          setOptions: jest.fn(),
          setParams: jest.fn(),
        },
      };
      const component = shallow(<Details {...props} />);
      expect(toJson(component)).toMatchSnapshot('Primary Details snapshot');
    });
    it('should match to snapshot - Secondary', () => {
      const props: Props = {
        route: {
          params: {
            isNew: false,
            item: {
              email: '',
              name: 'nome',
              phone: '',
              id: '1234',
              address: {
                cep: '',
                city: '14955-000',
                complement: '',
                neighborhood: '',
                number: '',
                state: '',
                street: '',
              },
            },
          },
          key: 'Details',
          name: 'Details',
        },
        navigation: {
          addListener: jest.fn(),
          canGoBack: jest.fn(),
          dangerouslyGetParent: jest.fn(),
          dangerouslyGetState: jest.fn(),
          dispatch: jest.fn(),
          goBack: jest.fn(),
          isFocused: jest.fn(),
          navigate: jest.fn(),
          pop: jest.fn(),
          popToTop: jest.fn(),
          push: jest.fn(),
          removeListener: jest.fn(),
          replace: jest.fn(),
          reset: jest.fn(),
          setOptions: jest.fn(),
          setParams: jest.fn(),
        },
      };
      const component = shallow(<Details {...props} />);
      expect(toJson(component)).toMatchSnapshot('Secondary Details snapshot');
    });
  });
});
