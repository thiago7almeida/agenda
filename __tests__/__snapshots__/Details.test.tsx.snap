// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details rendering should match to snapshot - Primary: Primary Details snapshot 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <TopNavigation
    accessoryLeft={[Function]}
    alignment="center"
    title="Detalhe do Contato"
  />
  <Divider />
  <Form
    isNew={true}
  />
</RNCSafeAreaView>
`;

exports[`Details rendering should match to snapshot - Secondary: Secondary Details snapshot 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <TopNavigation
    accessoryLeft={[Function]}
    alignment="center"
    title="Detalhe do Contato"
  />
  <Divider />
  <Form
    isNew={false}
    item={
      Object {
        "address": Object {
          "cep": "",
          "city": "14955-000",
          "complement": "",
          "neighborhood": "",
          "number": "",
          "state": "",
          "street": "",
        },
        "email": "",
        "id": "1234",
        "name": "nome",
        "phone": "",
      }
    }
  />
</RNCSafeAreaView>
`;
